apps/api/scripts/calculate-next-season-eligibility.ts

```diff
@@ -161,14 +148,14 @@ Examples:
         and(
           // Only claims with duration > 0 (actual stakes)
           gte(convictionClaims.duration, 1n),
-          // Only stakes from claims before the reference time
-          lte(convictionClaims.blockTimestamp, referenceTime),
-          // Ex-date calculation: blockTimestamp + duration seconds > referenceTime
-          sql`${convictionClaims.blockTimestamp} + (${convictionClaims.duration} * interval '1 second') > ${referenceTime}`,
+          // get stakes from claims before the end of the current season
+          lte(convictionClaims.blockTimestamp, currentSeason.endDate),
+          // and with durations that lasted at least until the start of the current season
+          sql`${convictionClaims.blockTimestamp} + (${convictionClaims.duration} * interval '1 second') > ${currentSeason.startDate}`,
```

Comment:
This doesn't make sense to me. I thought we want to find stakes with a block timestamp < the current season end and that stake start time + stake duration > current season end date (not start date).

Reply:
Fixed, good catch!